### PR TITLE
fix(#1377): add fullmenuwidth to goa-app-header - need code review

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
The bug is described at https://github.com/GovAlta/ui-components/issues/1377

Solution implemented: I add a new flag `fullmenuwidth` like Chris commented. https://github.com/GovAlta/ui-components/issues/1377#issuecomment-1930274360

As you can see in the video below, the screen is > 1024px (our breakpoint), and thanks to the added flag, I can make the menu to collapse like I want, depends on the context (menu collapse even when the screen is < 1700 based on my testing logic)
Screenshot/video: 

https://github.com/GovAlta/ui-components/assets/120135417/a090c1ae-b3d4-4f81-a594-7e1c51969fce

The code below for testing:

Angular:
```
<goa-two-column-layout maxcontentwidth="1400px" navcolumnwidth="30ch">
  <section slot="header">
    <goa-microsite-header
      type="alpha"
      level="alpha"
      version="UAT"
    ></goa-microsite-header>
    <goa-app-header heading="Design Systems" maxcontentwidth="1500px" [fullmenuwidth]="1700">
      <a href="/get-started">
       Get started
      </a>
      <a href="/patterns">Patterns and templates</a>
      <a href="/components">Components</a>
      <a href="/design-tokens">Styles</a>
      <a href="/content/capitalization">Content</a>
      <a href="/support">Support</a>
      <a href="https://github.com/GovAlta/ui-components/issues/new/choose" target="_blank">
      Report a bug
      </a>
      <a href="#">Support</a>
      <goa-app-header-menu heading="Tickets" leadingIcon="ticket">
        <a href="#">Cases</a>
        <a href="#">Payments</a>
        <a href="#">Outstanding</a>
      </goa-app-header-menu>
      <a href="#" className="interactive">Sign in</a>
    </goa-app-header>
  </section>
  <section slot="nav">
    <goa-side-menu>
      <goa-side-menu-group heading="Components">
        <a routerLink="/accordion"> Accordion </a>
         </goa-side-menu-group>
         </goa-side-menu>
  </section>
  Testing!
   <goa-app-footer slot="footer" maxcontentwidth="1400px"></goa-app-footer>
</goa-two-column-layout>
```

React Code: 

```
export function App() {
  return (
    <GoAOneColumnLayout>
      <section slot="header">
        <GoAMicrositeHeader type="alpha" version="UAT" />
        <GoAAppHeader heading="Design system" maxContentWidth="1500px" url={"/"} fullMenuWidth={1700}>
          <Link to="/get-started">Get started</Link>
          <Link to="/patterns">Patterns and templates</Link>
          <Link to="/components">Components</Link>
          <Link to="/design-tokens">Styles</Link>
          <Link to="/content/capitalization">Content</Link>
          <Link to="/support">Support</Link>
          <Link to="https://github.com/GovAlta/ui-components/issues/new/choose" target="_blank">
            Report a bug
          </Link>
        </GoAAppHeader>
      </section>
      <section slot="footer">
        <GoAAppFooter />
      </section>
    </GoAOneColumnLayout>
```

### CODE REVIEW

File: libs/web-components/src/components/app-header/AppHeader.svelte [hunk 1]
```diff
@@ -29,10 +31,11 @@
   $: _mobile = _windowWidth < MOBILE_BP;
   $: _tablet = _windowWidth >= MOBILE_BP && _windowWidth < TABLET_BP;
   $: _desktop = _windowWidth >= TABLET_BP;
+  $: _forceShowToggleMenu = toBoolean(forceshowtogglemenu);
   // @ts-expect-error
-  $: (async () => (_showToggleMenu = _desktop ? false : await hasChildren()))();
+ $: (async () => (_showToggleMenu = _desktop ? false : await hasChildren()))();
```
Comment (chrisolsen) [inline libs/web-components/src/components/app-header/AppHeader.svelte:36]: unwanted space removal, same with line 35

File: libs/web-components/src/components/app-header/AppHeader.svelte [hunk 2]
```diff
@@ -105,6 +108,9 @@
   data-testid={testid}
   style={`--max-content-width: ${maxcontentwidth || "100%"}`}
   class:show-menu={_showMenu}
+  class:mobile={_mobile || (_forceShowToggleMenu && _tablet)}
```
Comment (chrisolsen) [inline libs/web-components/src/components/app-header/AppHeader.svelte:111]: I think what I was suggesting earlier was to add an additional property that defines the *width* at which the menu collapse will occur at.

File: libs/react-components/src/lib/app-header/app-header.tsx [hunk 3]
```diff
@@ -17,6 +18,7 @@ export interface GoAAppHeaderProps {
   heading?: string;
   url?: string;
   maxContentWidth?: string;
+  fullMenuWidth?: number;
```
Comment (chrisolsen) [inline libs/react-components/src/lib/app-header/app-header.tsx:21]: I think having a prop name like `fullMenuBreakpoint` would make more sense.

**Review by ArakTaiRoth — APPROVED** [review]

**Review by chrisolsen — APPROVED** [review]

**Pull request merged**